File: backend/foxtrip-cart/src/main/java/haui/foxtrip/cart/service/CartService.java
package haui.foxtrip.cart.service;

import haui.foxtrip.cart.service.dto.CartResDTO;
import haui.foxtrip.cart.service.dto.UpsertCartItemReqDTO;
import java.util.UUID;

public interface CartService {

    CartResDTO getCart(UUID userId);

    CartResDTO upsertCartItem(UUID userId, UpsertCartItemReqDTO request);

    void clearCart(UUID userId);
}

File: front/src/services/api/cartService.js
import axiosClient from './axiosClient';

export const cartService = {
    getCart: () => {
        return axiosClient.get('/cart');
    },
    upsertItem: (tourId, quantity) => {
        return axiosClient.put('/cart/item', { tourId, quantity });
    },
    clearCart: () => {
        return axiosClient.delete('/cart');
    }
};
